File: src/main/java/src/main/java/org/example/a/test/People.java
package org.example.a.test;

import java.util.Objects;

public class People {
   String name;
   int age;

    People(String name, int age) {
        this.name = name;
        this.age = age;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null || getClass() != obj.getClass()) {
            return  false;
        }
        People people = (People) obj;
        return age == people.age && Objects.equals(name, people.name);
    }

    public static void main(String[] args) {
        People one = new People (null, 21);
        People two = new People (null, 21);

        System.out.println(one.equals(two));
    }
}

File: src/main/kotlin/a/test/People.kt
package a.test


class People internal constructor(var name: String?, var age: Int) {
    override fun equals(obj: Any?): Boolean {
        if (this === obj) {
            return true
        }
        if (obj == null || javaClass != obj.javaClass) {
            return false
        }
        val people = obj as People
        return age == people.age && name == people.name
    }

        fun main(args: Array<String>) {
            val one = People(null, 21)
            val two = People(null, 21)
            println(one == two)
        }
    }

